src/lakefs_spec/util.py

```diff
@@ -4,13 +4,16 @@
 
 from __future__ import annotations
 
+import asyncio
+import functools
 import hashlib
 import os
 import re
-from typing import Any, Callable, Generator, Protocol
+from typing import Any, Callable, Coroutine, Generator, Protocol, TypeVar
 
 from lakefs_sdk import Pagination
 from lakefs_sdk import __version__ as __lakefs_sdk_version__
+from typing_extensions import ParamSpec
```

Comment:
This is Python>=3.10 territory, so we should mark our package as that. (It is about time, anyway.)

Reply:
That's why it's imported from `typing_extensions`, so we can keep 3.9-compatibility.